content/articles/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/index.md

```diff
@@ -0,0 +1,146 @@
+
+The above image shows the numbers of the points that mediapipe uses to refer to different points of the hand.
+### Creating a hand gesture volume controller
+First of all, we will prepare our workspace. Launch the pycharm app. Click on the create a new project. This is shown in the screenshot provided below.
+
+![Pycharm](/engineering-education/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/pycharm.png)
+
+Click on the create button on the window that appears next.
+
+We now need to install the libraries we did discuss above in our project. To do so, follow the following steps. To install numpy, open the terminal and type the following command. 
+```bash
+pip install numpy
+```
+
+![Terminal](/engineering-education/creating-a-hand-gesture-volume-controller-using-python-and-pycharm/terminal.png)
+
+The above screenshot shows where to type the command for numpy installation.
+
+Wait for a few seconds till numpy is installed successfully. We are going to repeat the same steps for the other libraries. We will only replace the command part. Use the following commands for the respective libraries.
+```bash
+ pip install opencv-python
+
+```
+
+ ```bash
+pip install mediapipe
+
+ ```
+ ```bash
+pip install pycaw
+ ```
+ ```bash
+pip install python-math
+ ```
+ ```bash
+ pip install comtypes
+ ```
+ ```bash
+ pip install ctypes
+ ```
+
+Now let us jump into coding. This is where the most fun is as we get hands-on.
+
+```python
+import cv2
+import mediapipe as mp
+from math import hypot
+from ctypes import cast, POINTER
+from comtypes import CLSCTX_ALL
+from pycaw.pycaw import AudioUtilities, IAudioEndpointVolume
+import numpy as np
+
+cap = cv2.VideoCapture(0)  # Getting video input from our primary camera.
+# Detecting, initializing and configuring the hands.
+mpHands = mp.solutions.hands
+hands = mpHands.Hands()
+mpDraw = mp.solutions.drawing_utils
+# Accessing the speaker through pycaw.
+devices = AudioUtilities.GetSpeakers()
+interface = devices.Activate(IAudioEndpointVolume._iid_, CLSCTX_ALL, None)
+volume = cast(interface, POINTER(IAudioEndpointVolume))
+
+volMin, volMax = volume.GetVolumeRange()[:2]
+
+while True:
+    success, img = cap.read()  # if the camera we have specified works, we will capture an image.
+    imgRGB = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)  # converting the image the camera captures to RGB Image.
+    results = hands.process(imgRGB)  # completes the processing of the converted image.
+
+    lmList = []  # creating an empty list.
+    if results.multi_hand_landmarks:  # A list of all the detected hands.
+        #  Creating a loop so we can the list above so as to get the corresponding flag bit of each hand.
+        for handlandmark in results.multi_hand_landmarks:
+            for id, lm in enumerate(handlandmark.landmark):  # We are adding a counter and then returning it.
+                # Finding the finger joints of the hand
+                h, w, _ = img.shape

```

Comment:
@leeseshawn Instead of pasting the whole code and writing the description using the comments, you can separate the code into small sections and use the comments to write complete sentence/paragraph describing what each section does. 

Please refer to this [tutorial](https://github.com/section-engineering-education/engineering-education/pull/5163/files) for reference